pkg/jobs/registry.go

```diff
@@ -1282,6 +1286,33 @@ func (r *Registry) adoptionDisabled(ctx context.Context) bool {
 	return false
 }
 
+// A single job should not toggle its idleness more than twice per-minute as it
+// is logged and may write to persisted job state in the future
+func (r *Registry) MarkIdle(job *Job, isIdle bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if aj, ok := r.mu.adoptedJobs[job.ID()]; ok {
+		payload := job.Payload()
+		jobType := payload.Type()
+		jm := r.metrics.JobMetrics[jobType]
+		if aj.isIdle != isIdle {
+			log.Infof(r.serverCtx, "%s job %d: toggling idleness to %+v", jobType, job.ID(), isIdle)
+			if isIdle {
```

Comment:
This seems wrong to me: you probably should not increment multiple times even if MarkIdle(true) called multiple times.  And decremented below 0 would also panic.  I suggest we add a test for exactly this type of behavior.

Reply:
This should already function that way since it'll only inc/dec if `aj.isIdle != isIdle`, so multiple `MarkIdle(true)` fail that.  I'll add the below-0 error and a test around this :+1: